presto-product-tests/src/main/java/io/prestosql/tests/hive/TestInsertIntoHiveTable.java

```diff
@@ -146,4 +147,17 @@ public void testInsertIntoPartitionedWithSerdeProperty()
         assertThat(query("INSERT INTO " + tableNameInDatabase + " SELECT 1, 'presto', '2018-01-01'")).containsExactly(row(1));
         assertThat(query("SELECT * FROM " + tableNameInDatabase)).containsExactly(row(1, "presto", "2018-01-01"));
     }
+
+    @Test
+    public void testInsertIntoTransactionalTable()
+    {
+        onHive().executeQuery("" +
+                "CREATE TABLE test_insert_into_transactional_table(a bigint)" +
+                "CLUSTERED BY(a) INTO 4 BUCKETS STORED AS ORC TBLPROPERTIES ('transactional'='true', 'bucketing_version'='1')");
+
+        assertThat(() -> query("INSERT INTO test_insert_into_transactional_table (a) VALUES (42)"))
+                .failsWithMessage("Inserting into Hive transactional tables is not supported: default.test_insert_into_transactional_table");
+
+        onHive().executeQuery("DROP TABLE test_insert_into_transactional_table");
```

Comment:
try finally?

You could also check that the table doesn't have any data in Hive and is queryable

Reply:
> try finally?

We often don't `finally`. I know `finally` is more correct, but reduces readability a bit.

> You could also check that the table doesn't have any data in Hive

The exc is raised during planning, so we couldn't have written anything.

> and is queryable

It's not -- yet.
Maybe we don't have a check yet, I will add.